### PR TITLE
ieee802154: update KW41Z driver to also support KW40Z

### PR DESCRIPTION
**Pull request by bogdan-davidoaia**

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]

File: drivers/ieee802154/ieee802154_kw41z.c [hunk 1]
```diff
@@ -313,6 +323,9 @@ static inline void kw41z_rx(struct kw41z_context *kw41z, u8_t len)
 
 	net_pkt_frag_insert(pkt, frag);
 
+#if CONFIG_SOC_MKW41Z4
+	u16_t reg_val = 0;
```
Comment (tbursztyka) [inline drivers/ieee802154/ieee802154_kw41z.c]: looks like you could put this declaration at the beginning of the for loop. It's not super clean to declare a variable elsewhere than in the beginning of a code block

File: drivers/ieee802154/ieee802154_kw41z.c [hunk 2]
```diff
@@ -322,6 +335,27 @@ static inline void kw41z_rx(struct kw41z_context *kw41z, u8_t len)
 			frag->data[i] = reg_val >> 8;
 		}
 	}
+#else /* CONFIG_SOC_MKW40Z4 */
+	u32_t reg_val = 0;
```
Comment (tbursztyka) [inline drivers/ieee802154/ieee802154_kw41z.c]: same here

**Review by tbursztyka — CHANGES_REQUESTED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — CHANGES_REQUESTED** [review]
Weird that github takes the last review as the whole, I though it could do it commit by commit. Anyway, switching to request changes (a tiny one)

File: drivers/ieee802154/ieee802154_kw41z.c [hunk 3]
```diff
@@ -322,6 +335,27 @@ static inline void kw41z_rx(struct kw41z_context *kw41z, u8_t len)
...
+
+	/* PKT_BUFFER_RX needs to be accessed alligned to 32 bits */
+	for (i = 0; i < pkt_len; i++) {
+		switch (i % 4) {
+		case 0:
+			reg_val = *(((u32_t *)ZLL->PKT_BUFFER) + i/4);
```
Comment (MaureenHelm) [inline drivers/ieee802154/ieee802154_kw41z.c]: I know you're following the pattern from KW41 above, but is there a reason you didn't implement it like this:
```
reg_val = ZLL->PKT_BUFFER[i/4];
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/ieee802154/ieee802154_kw41z.c (hunk 1 above)
Comment (bogdan-davidoaia) [inline drivers/ieee802154/ieee802154_kw41z.c]: Done. I've declared and initialized in the for loop.

File: ext/hal/nxp/mcux/middleware/wireless/framework_5.3.3/XCVR/MKW40Z4/Makefile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2017, Linaro Ltd.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+obj-y += ../../OSAbstraction/Source/fsl_os_abstraction_zephyr.o
```
Comment (bogdan-davidoaia) [inline ext/hal/nxp/mcux/middleware/wireless/framework_5.3.3/XCVR/MKW40Z4/Makefile]: Done

File: ext/hal/nxp/mcux/middleware/wireless/framework_5.3.3/XCVR/Makefile [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2017, Linaro Ltd.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+MCUX_DEVICE = $(shell echo $(CONFIG_SOC) | tr '[:lower:]' '[:upper:]')
```
Comment (bogdan-davidoaia) [inline ext/hal/nxp/mcux/middleware/wireless/framework_5.3.3/XCVR/Makefile]: Done

File: drivers/ieee802154/ieee802154_kw41z.c (hunk 3 above)
Comment (bogdan-davidoaia) [inline drivers/ieee802154/ieee802154_kw41z.c]: Good point. Also adding a patch to change it for KW41Z.

**Review by tbursztyka — APPROVED** [review]

**Comment by MaureenHelm** [comment]
I see a lot of warnings around BBF_STEP_RECIP_def_c when I build this. I don't remember seeing them in your previous revision, but it's hard to see what's different between revisions of a PR. This is one thing I liked about gerrit that we don't have in GitHub.

**Comment by bogdan-davidoaia** [comment]
The warnings were there before. They all come from the XCVR driver imported from KSDK (which isn't really in line with the Zephyr coding style any way). I didn't address them as I didn't want to do too many changes in the imported code.

Should I add a patch to fix the warnings?

**Comment by jukkar** [comment]
This patch has been pending long time, @MaureenHelm do you wish that that @bogdan-davidoaia should fix the warnings?

**Comment by MaureenHelm** [comment]
> This patch has been pending long time, @MaureenHelm do you wish that that @bogdan-davidoaia should fix the warnings?

No, let's leave them alone. I agree with @bogdan-davidoaia 's initial assessment that it would be a lot of changes to imported code.

**Comment by jukkar** [comment]
@MaureenHelm Can you approve what is currently there so we can merge this PR, or is there still some changes needed?

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**